api/transport/header.go

```diff
@@ -102,6 +111,20 @@ func (h Headers) Items() map[string]string {
 	return h.items
 }
 
+// ForwardingItems returns the non-canonicalized version of the underlying map
+// for this Headers object. The returned map MUST NOT be changed.
```

Comment:
The MUST NOT clause is copy pasta. This does not apply since it returns a snapshot.